examples/cli-app/src/backend.rs

```diff
@@ -0,0 +1,67 @@
+// Interaction with bluetooth
+
+use std::sync::Mutex;
+use arctic::{EventHandler, PmdRead, PolarSensor};
+use tokio::sync::watch;
+use std::{fs::{OpenOptions, File}, io::Write, io};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+struct Handler {
+    rx: watch::Receiver<bool>,
+    output: Mutex<File>,
+}
+
+#[arctic::async_trait]
+impl EventHandler for Handler {
+    async fn measurement_update(&self, _ctx: &PolarSensor, data: PmdRead) {
+        let msg = format!("{:?}\n", data.data());
+        let mut output = self.output.lock().unwrap();
+        let _ = output.write_all(msg.as_bytes());
+        let _ = output.flush();
+    }
+
+    async fn should_continue(&self) -> bool {
+        *self.rx.borrow()
+    }
+} 
+
+impl Handler {
+    pub async fn new(rx: watch::Receiver<bool>) -> Result<Handler, io::Error> {
+        let mut output = String::from("cli-app/output/hr");
+        output.push_str(
+            &SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Error getting time")
+                .as_secs()
+                .to_string(),
+        );
+        output.push_str(".txt");
+
+        let output = OpenOptions::new().append(true).create(true).open(output).unwrap();
```

Comment:
I get a panic here:

> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 3, kind: NotFound, message: "The system cannot find the path specified." }', (...)\arctic\examples\cli-app\src\backend.rs:40:80

Issues seems to be I'm running the example from the example folder so `cli-app/output` should just be `output`. Not sure if the intention is to only run these from the top level.

Reply:
The examples were meant to be ran from the examples folder. I'll just add some error handling to find the output directory whether you run it from the project folder or examples folder.